src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -948,4 +966,47 @@ private class WaitForIndexItem
         }
     }
 
+    public class MergedBatchCommandDto : TransactionOperationsMerger.IReplayableCommandDto<BatchHandler.MergedBatchCommand>
+    {
+        public BatchRequestParser.CommandData[] ParsedCommands { get; set; }
+        public Queue<BatchHandler.MergedBatchCommand.AttachmentStream> AttachmentStreams;
+
+        public BatchHandler.MergedBatchCommand ToCommand(DocumentsOperationContext context, DocumentDatabase database)
+        {
+            if (ParsedCommands == null || ParsedCommands.Any() == false)
+            {
+                //Todo To check if empty array can append & if it is make sense to check it here 
```

Comment:
You don't need the check, we should never get an empty array here.
If we do get it, however, just handle it normally.

Reply:
done